pyaedt/modeler/circuits/PrimitivesNexxim.py

```diff
@@ -865,6 +865,55 @@ def create_voltage_pulse(self, compname=None, value_lists=[], location=[], angle
 
         return cmpid
 
+    @pyaedt_function_handler()
+    def create_voltage_pwl(
+        self, compname=None, time_list=[0], voltage_list=[0], location=[], angle=0, use_instance_id_netlist=False
+    ):
+        """Create a pwl voltage source.
+
+        Parameters
+        ----------
+        compname : str, optional
+            Name of the voltage pulse. The default is ``None``.
+        time_list : list, optional
+            List of time points for the pwl voltage source. The default is ``[0]``.
+        voltage_list : list, optional
+            List of voltages for the pwl voltage source. The default is ``[0]``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list.
+            The default is ``False``.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.cad.object3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+        >>> oEditor.CreateComponent
+        """
+        cmpid = self.create_component(
+            compname,
+            component_library="Independent Sources",
+            component_name="V_PWL",
+            location=location,
+            angle=angle,
+            use_instance_id_netlist=use_instance_id_netlist,
+        )
+
+        if len(time_list) != len(voltage_list):
+            raise ValueError("Length of time is not equal to length of voltage")
```

Comment:
Minor comment: to my knowledge, the current way of doing things in pyaedt is to return `False` or `None` when something goes wrong. @Samuelopez-ansys this line should return either oh them right ?
Note that I think we should raise Exception instead of returning None / False but I think that this should be done in some refactoring of the whole code base. IMO it would be better to be consistent here.

Reply:
Right, @boyang2022  please return False in this case, and also add a logger.error with the message.